File: framework/js-ts/testing/book-test-driven-development-by-madvirus/chap02/src/PasswordStrengthMeter.test.ts
import {
  PasswordStrength,
  PasswordStrengthMeter,
} from "./PasswordStrenthMeter";

test("모든 조건을 충족하면 암호 강도는 강함이어야 함", () => {
  const passwordStrengthMeter = new PasswordStrengthMeter();
  expect(passwordStrengthMeter.meter("ab12!@AB")).toBe(PasswordStrength.STRONG);
});

File: language/java-kotlin/book-tdd-by-javacan/src/test/java/chap02/PasswordStrengthMeterTest.java
package chap02;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.assertEquals;

public class PasswordStrengthMeterTest {
    @Test
    void meetAllCriteriaThenStrong() {
        PasswordStrengthMeter meter = new PasswordStrengthMeter();

        PasswordStrength result = meter.meter("ab12!@AB");
        assertEquals(PasswordStrength.STRONG, result);

        PasswordStrength result2 = meter.meter("abc1!Add");
        assertEquals(PasswordStrength.STRONG, result2);
    }
}
